### ISSUE TITLE
Getting- 'ReferenceError: navigator is not defined' on importing 'shaka-player'

### ISSUE THREAD

**Issue by devjyotik200** [shaka-project/shaka-player#5313]
I am using the shaka-player version 4.3.6 and i am trying to import it using `import shaka from 'shaka-player/dist/shaka-player.ui'.`

However i am getting the Navigator not defined error.
PS:- I am already using this component in client side.

Any idea how to resolve this?

<img width="1728" alt="Screenshot 2023-06-15 at 6 55 02 PM" src="https://github.com/shaka-project/shaka-player/assets/59660191/2d750fa0-5b82-4760-94c2-a95e21b5ab04">

**Comment by WINOFFRG** [comment]
This error only comes up when the component is being loaded on Server Side. You must make sure SSR is not being done here, Refer this https://github.com/WINOFFRG/limeplay/blob/main/home/src/components/Homepage/Limeplay.tsx#L5-L7
Import it as shown in above URL.

**Comment by devjyotik200** [comment]
@WINOFFRG yeah this is exactly what i did, but it didn't solve my problem. Apparently when i moved the video player component file to some other folder its working fine. Got the solution but not the RCA.

**Comment by WINOFFRG** [comment]
There can be multiple possibilities in your case, NextJS version 13 also has feature of React Server Components. If changing the file worked this could most probably be the issue https://nextjs.org/docs/getting-started/react-essentials otherwise I would recommend you to seek help from React/Next community itself as this issue isn't Shaka specific.

**Comment by devjyotik200** [comment]
Okay, yeah that maybe the cause. We are using NextJS version 13.2.4 .
Also @WINOFFRG  as you have already developed a react video player using shaka-player can you tell me whats the best way to change the UI of the player and add custom elements?
